test/test_unary_ufuncs.py

```diff
@@ -300,6 +304,16 @@ def test_reference_numerics(self, device, dtype, op):
 
     # Tests for testing (dis)contiguity consistency
 
+    def _multiple_output_check_helper(self,
```

Comment:
This is a really cool helper function for testing but we should consider whether the added complexity is worthwhile. There's probably some other ways to write this but let's focus on the rest of the PR first.